src/compass/utils/stitching/stitch_burst.py

```diff
@@ -0,0 +1,404 @@
+
+    Parameters:
+    -----------
+    polygons: list
+       List of shapely polygons to intersect
+    epsgs: list
+       List of EPSGs associate to 'polygons'
+
+    Returns:
+    -------
+    poly_int: shapely.Polygon
+       Result of polygon intersection
+    epsg_int: int
+       EPSG code associated to poly
+    '''
+    # Assert validity of inputs
+    assert (len(polygons) == len(epsgs))
+
+    # Initialize polygon and epsg of intersection
+    poly_int = polygons[0]
+    epsg_int = epsgs[0]
+
+    # Initialize coordinate transformation in case
+    # there are polygons with different epsgs
+    llh = ogr.osr.SpatialReference()
+    llh.ImportFromEPSG(epsg_int)
+    tgt = ogr.osr.SpatialReference()
+
+    for poly, epsg in zip(polygons, epsgs):
+        if epsg != epsg_int:
+            # Transform polygons in same coord system as epsg_int
+            tgt_x, tgt_y = [], []
+            x, y = poly.exterior.coords.xy
+            tgt.ImportFromEPSG(epsg)
+            trans = ogr.osr.CoordinateTransformation(llh, tgt)
+            for lx, ly in zip(x, y):
+                dummy_x, dummy_y, dummy_z = trans.Transform(lx, ly, 0)
+                tgt_x.append(dummy_x)
+                tgt_y.append(dummy_y)
+            poly = Polygon(list(zip(tgt_x, tgt_y)))
+        poly_int = poly.intersection(poly_int)
+
+    # To be conservative, apply some margin to the polygon (TO DO:
+    # check eps)
+    poly_int = poly_int.buffer(-margin)
+    return poly_int, epsg_int
+
+
+def get_stitching_dict(indir):
+    '''
+    Collect info on bursts to stitch and store them
+    in a panda dataframe
+
+    Parameters:
+    ----------
+    indir: str
+       Directory where bursts are stored (organized by date)
+
+    Returns:
+    -------
+    cfg: pandas.DataFrame
+       Dataframe collecting info on bursts to stitch
+    '''
+    # Create dictionary where to store results
+    cfg = {'burst_id': [], 'granule_id': [], 'polygon': [],
+           'date': [], 'epsg': []}
+    # Get list of directory under dir_list
+    dir_list = os.listdir(indir)
+    for dir in dir_list:
+        # List metadata files in the directory
+        meta_list = sorted(glob.glob(f'{indir}/{dir}/*json'))
+        for path in meta_list:
+            # Read metadata file
+            metadata = read_metadata(path)
+            # Read info and store in dictionary
+            cfg['burst_id'].append(get_metadata(metadata, 'burst_id'))
+            datestr = get_metadata(metadata, 'sensing_start')
+            date = datetime.fromisoformat(datestr).strftime("%Y%m%d")
+            filename = f"{get_metadata(metadata, 'burst_id')}_{date}_VV.slc"
+            cfg['granule_id'].append(f'{indir}/{dir}/{filename}')
+            poly = get_metadata(metadata, 'border')
+            cfg['polygon'].append(shapely.wkt.loads(poly))
+            cfg['date'].append(date)
+            geogrid = get_metadata(metadata, 'geogrid')
+            cfg['epsg'].append(geogrid['epsg'])
```

Comment:
This could probably be faster by filtering by file name ([file name includes `burst_id`](https://github.com/opera-adt/COMPASS/blob/6d5f80247a0baf1b146673471543cb34dc7ac6e2/src/compass/geo_cslc.py#L125)) and skipping the file loading/reading [here](https://github.com/opera-adt/COMPASS/blob/6d5f80247a0baf1b146673471543cb34dc7ac6e2/src/compass/utils/stitching/stitch_burst.py#L292). e.g.
```python
        for path in meta_list:
            # Check if burst_id wanted
            if path.split('_')[0] not in wanted_burst_ids:
                continue
```